firebaseai/testapp/Assets/Firebase/Sample/FirebaseAI/UIHandlerAutomated.cs

```diff
@@ -1026,6 +1069,113 @@ async Task InternalTestFunctionCallWithArguments() {
+
+      AssertEq("GroundingChunks count", grounding.GroundingChunks.Count(), 2);
+      var firstChunk = grounding.GroundingChunks.First();
+      Assert("GroundingChunk.Web should not be null", firstChunk.Web.HasValue);
+      var webChunk = firstChunk.Web.Value;
+      AssertEq("WebGroundingChunk.Title", webChunk.Title, "accuweather.com");
+      Assert("WebGroundingChunk.Uri should not be null", webChunk.Uri != null);
+      Assert("WebGroundingChunk.Domain should be null or empty", string.IsNullOrEmpty(webChunk.Domain));
+
+      AssertEq("GroundingSupports count", grounding.GroundingSupports.Count(), 3);
+      var firstSupport = grounding.GroundingSupports.First();
+      var segment = firstSupport.Segment;
+      AssertEq("Segment.Text", segment.Text, "The current weather in London, United Kingdom is cloudy.");
+      AssertEq("Segment.StartIndex", segment.StartIndex, 0);
+      AssertEq("Segment.PartIndex", segment.PartIndex, 0);
+      AssertEq("Segment.EndIndex", segment.EndIndex, 56);
+      AssertEq("GroundingChunkIndices count", firstSupport.GroundingChunkIndices.Count(), 1);
+      AssertEq("GroundingChunkIndices content", firstSupport.GroundingChunkIndices.First(), 0);
+    }
+
+    // Test that parsing a Google AI response with empty GroundingChunks works.
+    // https://github.com/FirebaseExtended/vertexai-sdk-test-data/blob/main/mock-responses/googleai/unary-success-google-search-grounding-empty-grounding-chunks.json
+    async Task InternalTestGoogleAIGroundingEmptyChunks() {
+      Dictionary<string, object> json = await GetGoogleAIJsonTestData("unary-success-google-search-grounding-empty-grounding-chunks.json");
+      GenerateContentResponse response = GenerateContentResponse.FromJson(json, FirebaseAI.Backend.InternalProvider.GoogleAI);
+
+      Assert("Response missing candidates.", response.Candidates.Any());
+      var candidate = response.Candidates.First();
+      Assert("Candidate should have GroundingMetadata", candidate.GroundingMetadata.HasValue);
+
+      var grounding = candidate.GroundingMetadata.Value;
+      AssertEq("WebSearchQueries count", grounding.WebSearchQueries.Count(), 1);
+      AssertEq("GroundingChunks count", grounding.GroundingChunks.Count(), 2);
+      Assert("First GroundingChunk.Web should be null", !grounding.GroundingChunks.ElementAt(0).Web.HasValue);
+      Assert("Second GroundingChunk.Web should be null", !grounding.GroundingChunks.ElementAt(1).Web.HasValue);
+
+      AssertEq("GroundingSupports count", grounding.GroundingSupports.Count(), 1);
+      var support = grounding.GroundingSupports.First();
+      AssertEq(
+          "Segment.Text",
+          support.Segment.Text,
+          "There is a 0% chance of rain and the humidity is around 41%.");
+    }
+
+    // Test parsing an empty GroundingMetadata object.
+    async Task InternalTestGroundingMetadata_Empty() {
```

Comment:
For this and the one below, since you aren't using 'await' in the function it'll print a warning.  So you can remove the 'async' keyword, and return 'Task.CompletedTask' at the end of each.